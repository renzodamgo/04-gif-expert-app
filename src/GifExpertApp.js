export const GifExpertApp = () => {
    const categories = ['One punch man','Naruto','Dragon Ball']
    return (
        <div>
            <h2>GifExpertApp</h2>
            <hr/>
        </div>
    )
}
