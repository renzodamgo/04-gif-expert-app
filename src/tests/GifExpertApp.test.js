import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Pruebas den GifExpertApp', () => {
    test('Debería hacer match con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })
    test('Debería mostarse un arreeglo de categorías', () => {
        const categories = ['One Punch Man', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GiftGrid').length).toBe(categories.length)
    })
})