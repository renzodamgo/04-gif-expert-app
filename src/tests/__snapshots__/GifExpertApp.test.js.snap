// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas den GifExpertApp Debería hacer match con el snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas den GifExpertApp Debería mostarse un arreeglo de categorías 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="One Punch Man"
      key="One Punch Man"
    />
    <GiftGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
